scrapy/utils/iterators.py

```diff
@@ -54,12 +75,20 @@ def xmliter(obj, nodename):
         yield Selector(text=nodetext, type="xml")
 
 
-def xmliter_lxml(obj, nodename, namespace=None, prefix="x"):
+def xmliter_lxml(
+    obj: Union[Response, str, bytes],
+    nodename: str,
+    namespace: Optional[str] = None,
+    prefix: str = "x",
+) -> Generator[Selector, Any, None]:
     from lxml import etree
 
     reader = _StreamReader(obj)
     tag = f"{{{namespace}}}{nodename}" if namespace else nodename
-    iterable = etree.iterparse(reader, tag=tag, encoding=reader.encoding)
+    # technically, etree.iterparse only needs .read() AFAICS, but this is how it's typed
+    iterable = etree.iterparse(
+        cast("SupportsReadClose[bytes]", reader), tag=tag, encoding=reader.encoding
```

Comment:
:lipstick: Would it work to move the type hint to line 60/86 above instead of casting here?

PS: I’m not sure we need the comment.

Reply:
Casting hides `reader.encoding`.